back/app/services/invites/seats_counter.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Invites::SeatsCounter
+  def count_new_seats(&_block)
+    new_admins_number = nil
+    new_project_moderators_number = nil
+    User.transaction do
+      yield
+      new_admins_number = User.billed_admins.count
+      new_project_moderators_number = User.billed_moderators.count
+      # https://api.rubyonrails.org/classes/ActiveRecord/Rollback.html
+      raise ActiveRecord::Rollback
+    end
+
+    {
+      newly_added_admins_number: new_admins_number - User.billed_admins.count,
+      newly_added_project_moderators_number: new_project_moderators_number - User.billed_moderators.count
```

Comment:
If the transaction fails or is rolled back, will the values on lines 16 & 17 be negative? Could this be a problem?

Reply:
Very good observation. If the transaction is aborted by `ActiveRecord::Rollback` (our normal flow), the code just continues to execute and the method returns a hash. But if the transaction is aborted by any other error, this error is not swallowed by `transaction do` block and so it's raised further (so, the hash `{ newly_added_admins_number: ...` is not created.

```ruby
[37] pry(main)> User.transaction do
  raise ActiveRecord::Rollback
end
=> nil
[38] pry(main)> User.transaction do
  raise RuntimeError
end
RuntimeError: RuntimeError
```

Reply:
So, `new_admins_number` can be nil only if a hash `{ newly_added_admins_number: ...` is not created.

Reply:
I see. Thanks for the explanation. 👍